planner/core/planbuilder.go

```diff
@@ -3722,6 +3726,39 @@ func (b *PlanBuilder) buildValuesListOfInsert(ctx context.Context, insert *ast.I
 	return nil
 }
 
+func (b *PlanBuilder) checkSubQueryNameOfInsert(ctx context.Context, insert *ast.InsertStmt, srcTbl *ast.TableName) error {
```

Comment:
I think it can be done in `getInsertColExpr`, specifically the last hook func argument of `rewriteWithPreprocess` invocation. Then it will be much simpler than messy `checkSubQueryNameOfInsert`, which is complicated/slower.